{
  "issue_number": 8329,
  "title": "Flaky test: raftstore::test_lease_read::test_node_batch_id_in_lease",
  "body": "## Bug Report\r\n\r\n<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->\r\n\r\n### What version of TiKV are you using?\r\nmaster\r\n\r\n### What operating system and CPU are you using?\r\njenkins\r\n\r\n### Steps to reproduce\r\n\r\nThe mentioned test is failing on master in a couple of PRs, including:\r\nhttps://github.com/tikv/tikv/pull/8321\r\nhttps://github.com/tikv/tikv/pull/8235\r\nhttps://github.com/tikv/tikv/pull/8330\r\n\r\nCan't reproduce on local yet.\r\n\r\n```\r\n[2020-07-24T07:30:33.545Z] test config::dynamic::gc_worker::test_change_io_limit_by_config_manager ... ok\r\n[2020-07-24T07:30:36.056Z] test pd::test_rpc_client::test_retry_rpc_client ... ok\r\n[2020-07-24T07:30:46.014Z] test raftstore::test_lease_read::test_node_batch_id_in_lease ... thread 'main' panicked at 'assertion failed: std::ptr::eq(snaps2[0].get_snapshot(), snaps2[2].get_snapshot())', tests/integrations/raftstore/test_lease_read.rs:371:5\r\n[2020-07-24T07:30:46.014Z] stack backtrace:\r\n[2020-07-24T07:30:46.597Z]    0: backtrace::backtrace::libunwind::trace\r\n[2020-07-24T07:30:46.597Z]              at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86\r\n[2020-07-24T07:30:46.597Z]    1: backtrace::backtrace::trace_unsynchronized\r\n[2020-07-24T07:30:46.597Z]              at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66\r\n[2020-07-24T07:30:46.597Z]    2: std::sys_common::backtrace::_print_fmt\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/sys_common/backtrace.rs:78\r\n[2020-07-24T07:30:46.597Z]    3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/sys_common/backtrace.rs:59\r\n[2020-07-24T07:30:46.597Z]    4: core::fmt::write\r\n[2020-07-24T07:30:46.597Z]              at src/libcore/fmt/mod.rs:1076\r\n[2020-07-24T07:30:46.597Z]    5: std::io::Write::write_fmt\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/io/mod.rs:1537\r\n[2020-07-24T07:30:46.597Z]    6: std::sys_common::backtrace::_print\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/sys_common/backtrace.rs:62\r\n[2020-07-24T07:30:46.597Z]    7: std::sys_common::backtrace::print\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/sys_common/backtrace.rs:49\r\n[2020-07-24T07:30:46.597Z]    8: std::panicking::default_hook::{{closure}}\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/panicking.rs:198\r\n[2020-07-24T07:30:46.597Z]    9: std::panicking::default_hook\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/panicking.rs:217\r\n[2020-07-24T07:30:46.597Z]   10: std::panicking::rust_panic_with_hook\r\n[2020-07-24T07:30:46.597Z]              at src/libstd/panicking.rs:520\r\n[2020-07-24T07:30:46.597Z]   11: std::panicking::begin_panic\r\n[2020-07-24T07:30:46.597Z]              at /rustc/16957bd4d3a5377263f76ed74c572aad8e4b7e59/src/libstd/panicking.rs:450\r\n[2020-07-24T07:30:46.597Z]   12: integrations::raftstore::test_lease_read::test_batch_id_in_lease\r\n[2020-07-24T07:30:46.597Z]              at tests/integrations/raftstore/test_lease_read.rs:371\r\n[2020-07-24T07:30:46.597Z]   13: integrations::raftstore::test_lease_read::test_node_batch_id_in_lease\r\n[2020-07-24T07:30:46.597Z]              at tests/integrations/raftstore/test_lease_read.rs:275\r\n[2020-07-24T07:30:46.597Z]   14: integrations::raftstore::test_lease_read::test_node_batch_id_in_lease::{{closure}}\r\n[2020-07-24T07:30:46.597Z]              at tests/integrations/raftstore/test_lease_read.rs:272\r\n[2020-07-24T07:30:46.597Z]   15: core::ops::function::FnOnce::call_once\r\n[2020-07-24T07:30:46.597Z]              at /rustc/16957bd4d3a5377263f76ed74c572aad8e4b7e59/src/libcore/ops/function.rs:233\r\n[2020-07-24T07:30:46.597Z]   16: test_util::runner::run_test_with_hook::{{closure}}::{{closure}}\r\n[2020-07-24T07:30:46.597Z]              at components/test_util/src/runner.rs:64\r\n[2020-07-24T07:30:46.597Z]   17: core::ops::function::FnOnce::call_once{{vtable.shim}}\r\n[2020-07-24T07:30:46.597Z]              at /rustc/16957bd4d3a5377263f76ed74c572aad8e4b7e59/src/libcore/ops/function.rs:233\r\n[2020-07-24T07:30:46.597Z]   18: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once\r\n[2020-07-24T07:30:46.597Z]              at /rustc/16957bd4d3a5377263f76ed74c572aad8e4b7e59/src/liballoc/boxed.rs:1078\r\n[2020-07-24T07:30:46.597Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.\r\n[2020-07-24T07:30:46.597Z] FAILED\r\n[2020-07-24T07:30:56.597Z] test raftstore::test_multi::test_multi_server_leader_crash ... ok\r\n[2020-07-24T07:31:03.209Z] test raftstore::test_replication_mode::test_migrate_replication_mode ... ok\r\n[2020-07-24T07:31:15.602Z] test raftstore::test_tombstone::test_server_readd_peer ... ok\r\n[2020-07-24T07:31:16.978Z] test storage::test_storage::bench_txn_store_rocksdb_put_x100 ... ok\r\n[2020-07-24T07:31:16.978Z] \r\n[2020-07-24T07:31:16.978Z] failures:\r\n[2020-07-24T07:31:16.978Z] \r\n[2020-07-24T07:31:16.978Z] failures:\r\n[2020-07-24T07:31:16.978Z]     raftstore::test_lease_read::test_node_batch_id_in_lease\r\n[2020-07-24T07:31:16.978Z] \r\n[2020-07-24T07:31:16.978Z] test result: FAILED. 6 passed; 1 failed; 0 ignored; 0 measured; 346 filtered out\r\n```\r\n",
  "state": "closed",
  "created_at": "2020-07-24T23:10:09Z",
  "updated_at": "2020-08-10T03:17:13Z",
  "closed_at": "2020-07-29T10:38:33Z",
  "labels": [
    "type/bug",
    "severity/minor"
  ],
  "comments_data": []
}